std/random.d

```diff
@@ -2634,6 +2634,103 @@ do
     assert(i == 0);
 }
 
+/+ @nogc bool array designed for RandomCover.
+- constructed with an invariable length
+- small length means 0 alloc and bit field (if up to 32(x86) or 64(x64) choices to cover)
+- bigger length means non-GC heap allocation(s) and dealloc. +/
+private struct RandomCoverChoices
+{
+    private void* buffer;
+    private immutable size_t _length;
+    private immutable bool hasPackedBits;
+
+    this(this) pure nothrow @nogc @trusted
+    {
+        import core.memory : pureMalloc;
+        import core.stdc.string : memmove;
+        import core.exception : onOutOfMemoryError;
+
+        if (!hasPackedBits && buffer !is null)
+        {
+            void* nbuffer = pureMalloc(_length);
+            if (nbuffer is null)
+                onOutOfMemoryError();
+            buffer = memmove(nbuffer, buffer, _length);
```

Comment:
You can use `memcpy` here instead of `memmove` because the destination and source cannot overlap.